### PR TITLE
[Autoscaler] Proper resource demand plumbing

### PR DESCRIPTION
**Pull request by wuisawesome**
## Why are these changes needed?

This properly plumbs resource demand metrics so that they don't conflict with the manual api. 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/autoscaler/autoscaler.py [hunk 1]
```diff
@@ -197,14 +197,19 @@ def _update(self):
...
+            resource_demand_vector = self.resource_demand_vector + \
+                self.load_metrics.get_resource_demand_vector()
+            if resource_demand_vector:
+                to_launch = (self.resource_demand_scheduler.get_nodes_to_launch(
+                    self.provider.non_terminated_nodes(tag_filters={}),
+                    self.pending_launches.breakdown(),
+                    resource_demand_vector))
+                # TODO(ekl) also enforce max launch concurrency here?
+                print(f"vector: {resource_demand_vector}")
+                print(f"result: {to_launch}")
```
Comment (ericl) [inline python/ray/autoscaler/autoscaler.py:210]: Please remove prints before merging.

File: python/ray/test_utils.py [hunk 2]
```diff
@@ -336,6 +336,25 @@ def dicts_equal(dict1, dict2, abs_tol=1e-4):
     return True
 
 
+def same_elements(elems_a, elems_b):
+    """Checks if two iterables (such as lists) contain the same elements. Elements
```
Comment (ericl) [inline python/ray/test_utils.py:340]: Can you instead check `set(a) == set(b)`?

**Review by ericl — APPROVED** [review]
Looks great

File: python/ray/test_utils.py (hunk 2 above)
Comment (wuisawesome) [reply python/ray/test_utils.py:340]: unfortunately can't because dictionaries aren't hashable

**Comment by ericl** [comment]
Seems like some failing tests / merge conflicts.

**Comment by ericl** [comment]
Hmm it seems a bunch of tests are now failing (test_stress, etc.) after https://travis-ci.com/github/ray-project/ray/jobs/378239045

I also see it failing on this PR.

**Pull request merged**